### PR TITLE
[BugFix] Fix hive decimal type error when query hive connector table

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #
https://github.com/StarRocks/starrocks/issues/6632

related with https://github.com/StarRocks/starrocks/issues/5062

## Problem Summary(Required) ：

support query decimal column of hive connector table

### CODE REVIEW

**Review by caneGuy — DISMISSED** [review]

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 10 / 39 (25.64%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|8|32|25.00%|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|2|4|50.00%|

**Review by imay — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java [hunk 1]
```diff
@@ -227,8 +227,8 @@ public boolean isRefreshColumn(Map<String, Column> preNameToColumn,
                     needRefreshColumn = true;
                     break;
                 }
-                PrimitiveType type = convertColumnType(fieldSchema.getType());
-                if (type != column.getType().getPrimitiveType()) {
+                ScalarType type = convertColumnType(fieldSchema.getType());
+                if (type != column.getType()) {
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java:231]: If this is ScalarType, `!=` means if they are the same object.
I think you want to call is `equals()`?

File: fe/fe-core/src/main/java/com/starrocks/external/hive/Utils.java [hunk 2]
```diff
@@ -143,4 +143,11 @@ public static String getTypeKeyword(String type) {
         }
         return keyword;
     }
+
+    // Decimal string like "Decimal(3,2)"
+    public static String[] getPrecisionAndScale(String typeStr) {
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/Utils.java:148]: why not just return `Int[]`

**Comment by DorianZheng** [comment]
https://cwiki.apache.org/confluence/display/hive/languagemanual+types#LanguageManualTypes-DecimalsdecimalDecimals

image

**Comment by DorianZheng** [comment]
I'm not sure if hms will return type `Decimal` only, could you verify it.

File: fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java (hunk 1 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java]: How stupid am I.

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java]: What we get is after hms has been normalized

**Comment by stephen-shelby** [comment]
> I'm not sure if hms will return type `Decimal` only, could you verify it.

What we get is after hms has been normalized

File: fe/fe-core/src/main/java/com/starrocks/external/hive/Utils.java [hunk 3]
```diff
@@ -143,4 +143,11 @@ public static String getTypeKeyword(String type) {
         }
         return keyword;
     }
+
+    // Decimal string like "Decimal(3,2)"
+    public static int[] getPrecisionAndScale(String typeStr) {
+        String partWithComma = typeStr.substring(8, typeStr.length() - 1);
```
Comment (satanson) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/Utils.java:149]: using a regex match group is more rebust

"decimal\s*\((\d+)\s*,\s*(\d+)\)"
group(1)->precision
group(2)->scale

Add UT for this function

**Review by DorianZheng — APPROVED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 14 / 38 (36.84%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|8|30|26.67%|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|2|4|50.00%|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|4|4|100.00%|

**Review by satanson — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**